analyzer/src/main/kotlin/PackageManagerFactory.kt

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright (c) 2017 HERE Europe B.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+package com.here.ort.analyzer
+
+import java.nio.file.FileSystems
+
+/**
+ * A factory to create new instances of [T]. It also stores some static information about the [PackageManager] type it
+ * creates.
+ *
+ *
+ * @property homepageUrl The URL to the package manager's homepage.
+ * @property primaryLanguage The name of the programming language this package manager is primarily used with.
+ * @param globsForDefinitionFiles A prioritized list of glob patterns of definition files supported by this package
+ *                                manager.
+ */
+abstract class PackageManagerFactory<out T : PackageManager>(
+        val homepageUrl: String,
+        val primaryLanguage: String,
+        globsForDefinitionFiles: List<String>
+) {
+    /**
+     * Creates a new instance of the [PackageManager].
+     */
+    abstract fun create(): T
+
+    fun name() = javaClass.name.toUpperCase().substringBefore('$').substringAfterLast('.')
```

Comment:
Please add some minimal docs so that everything in this class is documented.

Reply:
Done.